contentcuration/contentcuration/models.py

```diff
@@ -266,60 +268,48 @@ def generate_object_storage_name(checksum, filename):
 
 
 def generate_storage_url(filename, request=None, *args):
+    """
+    Generate a storage URL for the given content filename.
+    """
 
     path = generate_object_storage_name(os.path.splitext(filename)[0], filename)
 
-    # We have to handle these cases:
-    # 1. In normal kubernetes, nginx will proxy this for us. So just return a relative path.
-    # 2. When ran through Telepresence, the cluster's DNS host is available to the host. So we should then use whatever is
-    #    defined in AWS_S3_STORAGE_URL.
-    # 3. We go through nanobox, meaning it's the same IP address as the server, but a different port.
-    # New! 4. We go through a production object storage server. In that case, just return the object storage URL
+    # There are three scenarios where Studio might be run as:
+    #
+    # 1. In normal kubernetes, nginx will proxy for us. We'll know we're in kubernetes when the
+    # environment variable RUN_MODE=k8s
+    #
+    # 2. In Docker Compose and bare metal runserver, we'll be running in runserver, and minio
+    # will be exposed in port 9000 in the host's localhost network.
 
-    # host = get_local_ip_address()
-    # # assume the port is 9000, the default of minio
-    # port = 9000
+    # Note (aron): returning the true storage URL (e.g. https://storage.googleapis.com/storage/a.mp4)
+    # isn't too important, because we have CDN in front of our servers, so it should be cached.
+    # But change the logic here in case there is a potential for bandwidth and latency improvement.
 
     # Detect our current state first
-    IS_RUNSERVER = "runserver" in sys.argv
-    HAS_MINIO_DNS_RECORD = False
-    IS_PRODUCTION_OBJECT_STORAGE_SERVER = "https://" in settings.AWS_S3_ENDPOINT_URL
-
-    # if we're using production, just return the real path to the prod server
-
-    try:
-        HAS_MINIO_DNS_RECORD = socket.gethostbyname("minio")
-    except socket.gaierror:
-        logging.debug("minio DNS record not detected. Assuming we're in a plain runserver command without Kubernetes.")
+    run_mode = os.getenv("RUN_MODE")
```

Comment:
Is this something developers need to add to their environment variables?

Reply:
Nope! It has a default behvaiour that should work for non-production environments.